crates/ark/src/interface.rs

```diff
@@ -1025,6 +1078,61 @@ impl RMain {
         }
     }
 
+    fn handle_pending_line(&mut self, buf: *mut c_uchar, buflen: c_int) -> Option<ConsoleResult> {
+        if self.error_occurred {
+            // If an error has occurred, we've already sent a complete expression that resulted in
+            // an error. Flush the remaining lines and return to `read_console()`, who will handle
+            // that error.
+            self.pending_lines.clear();
+            return None;
+        }
+
+        let Some(input) = self.pending_lines.pop() else {
+            // No pending lines
+            return None;
+        };
+
+        Self::on_console_input(buf, buflen, input);
+        Some(ConsoleResult::NewInput)
+    }
+
+    fn check_console_input(input: &str) -> amalthea::Result<()> {
```

Comment:
I've reworked this method so it returns a `Result`. We now convert the `Err` case to a `ConsoleResult::Error` at the call site, which I think is clearer.

Reply:
Oh much nicer thanks